### PR TITLE
Laboratory work #1, Liya Kunchenko - 23FPL1

### PR DESCRIPTION
**Pull request by li-kuuu**

### CODE REVIEW

File: lab_1/Laboratory Work #1 Liya Kunchenko - 23FPL1 [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Lab 1.
+"""
+
+
+def say_hello_world() -> None:
+    print('Hello world!')
+
+
+say_hello_world()
```
Comment (EIzharikov) [inline lab_1/Laboratory Work #1 Liya Kunchenko - 23FPL1:9]: Revert this. You must have only 1 directory with lab and it's name is "lab_1_classify_by_unigrams". Artyom Mikhailovich will help you at the next seminar.

File: requirements.txt [hunk 2]
```diff
@@ -1 +1,2 @@
 
+
```
Comment (EIzharikov) [inline requirements.txt:2]: revert this too

**Review by EIzharikov — CHANGES_REQUESTED** [review]
Nothing to review, there are problems with folders, BUT creating a request is already a success! Good luck with your lab work.

**Comment by EIzharikov** [comment]
There's nothing to watch, I advise you to speed up! I wish you good luck!

File: lab_1/Laboratory Work #1 Liya Kunchenko - 23FPL1 (hunk 1 above)
Comment (EIzharikov) [reply lab_1/Laboratory Work #1 Liya Kunchenko - 23FPL1:9]: ?

File: lab_1_classify_by_unigrams/main.py [hunk 3]
```diff
@@ -20,6 +20,15 @@ def tokenize(text: str) -> list[str] | None:
 
     In case of corrupt input arguments, None is returned
     """
+    text_of_token = []
```
Comment (EIzharikov) [inline lab_1_classify_by_unigrams/main.py:23]: what if input is not string?

File: lab_1_classify_by_unigrams/main.py [hunk 4]
```diff
@@ -20,6 +20,15 @@ def tokenize(text: str) -> list[str] | None:
...
+    if isinstance(text, str):
+        text = text.lower()
+        for element in text:
+            if element == 'º':
```
Comment (EIzharikov) [inline lab_1_classify_by_unigrams/main.py:27]: for what purpose you have this?

File: lab_1_classify_by_unigrams/main.py [hunk 5]
```diff
@@ -34,7 +43,20 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
 
     In case of corrupt input arguments, None is returned
     """
-
+    if isinstance(tokens, list):
```
Comment (EIzharikov) [inline lab_1_classify_by_unigrams/main.py:46]: make early stop

File: lab_1_classify_by_unigrams/main.py [hunk 6]
```diff
@@ -34,7 +43,20 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
...
+        for element in tokens:
+            if isinstance(element, str):
```
Comment (EIzharikov) [inline lab_1_classify_by_unigrams/main.py:48]: the same. Check it earlier and swap the conditions, you shoud stop earlier if input is corrupted

File: lab_1_classify_by_unigrams/main.py [hunk 7]
```diff
@@ -49,6 +71,10 @@ def create_language_profile(language: str, text: str) -> dict[str, str | dict[st
 
     In case of corrupt input arguments, None is returned
     """
+    if isinstance(language, str):
+        if isinstance(text, str):
+            profile_of_language = {'name': language, 'freq': calculate_frequencies(tokenize(text))}
```
Comment (EIzharikov) [inline lab_1_classify_by_unigrams/main.py:76]: what if calculate_frequencies will return None?

File: lab_1_classify_by_unigrams/settings.json [hunk 8]
```diff
@@ -1,3 +1,3 @@
 {
-  "target_score": 0
+  "target_score": 6
```
Comment (EIzharikov) [inline lab_1_classify_by_unigrams/settings.json:2]: Nice aim!

File: lab_1_classify_by_unigrams/start.py [hunk 9]
```diff
@@ -1,22 +1,23 @@
+#Laboratory Work #1
```
Comment (EIzharikov) [inline lab_1_classify_by_unigrams/start.py:1]: Revert

File: lab_1_classify_by_unigrams/start.py [hunk 10]
```diff
@@ -1,22 +1,23 @@
...
 """
 Language detection starter
 """
 # pylint:disable=too-many-locals, unused-argument, unused-variable
 
-
```
Comment (EIzharikov) [inline lab_1_classify_by_unigrams/start.py:6]: revert

File: lab_1_classify_by_unigrams/start.py [hunk 11]
```diff
@@ -1,22 +1,23 @@
...
 def main() -> None:
     """
     Launches an implementation
     """
+    from main import create_language_profile
```
Comment (EIzharikov) [inline lab_1_classify_by_unigrams/start.py:11]: import at the start of this file

File: lab_1_classify_by_unigrams/start.py [hunk 12]
```diff
@@ -1,22 +1,23 @@
...
     with open("assets/texts/en.txt", "r", encoding="utf-8") as file_to_read_en:
         en_text = file_to_read_en.read()
     with open("assets/texts/de.txt", "r", encoding="utf-8") as file_to_read_de:
         de_text = file_to_read_de.read()
     with open("assets/texts/unknown.txt", "r", encoding="utf-8") as file_to_read_unk:
         unknown_text = file_to_read_unk.read()
-    result = None
+    result = create_language_profile('english', en_text)
+    print(result)
```
Comment (EIzharikov) [inline lab_1_classify_by_unigrams/start.py:19]: good

**Review by EIzharikov — COMMENTED** [review]
Nice progress, but fix all my comments and be ready for the next review. Good luck!

**Review by EIzharikov — COMMENTED** [review]
Nothing new to review, fix all my comments and answer to them. I wish you luck with lab work defending.

File: lab_1_classify_by_unigrams/main.py (hunk 3 above)
Comment (li-kuuu) [reply lab_1_classify_by_unigrams/main.py:25]: Done

File: lab_1_classify_by_unigrams/main.py (hunk 6 above)
Comment (li-kuuu) [reply lab_1_classify_by_unigrams/main.py]: Done

**Pull request closed without merging**